### ISSUE TITLE
integrate produces wrong answer for Gaussian Moments

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#1855]
Imported from SourceForge on 2024-07-05 09:25:43
Created by **[blackgriffen](https://sourceforge.net/u/blackgriffen/)** on 2014-04-28 02:56:22
Original: https://sourceforge.net/p/maxima/bugs/2726

---

~~~~
::text
integrate( exp( -x^2/2 ) / sqrt(2*%pi) * x^m, x, -0.25, 0.25 );
%, m = 4;
~~~~

The result comes out negative, and it should be positive. Maxima produces the correct result if the value of m is inserted before doing the integral.

On: Maxima 5.33.0 http://maxima.sourceforge.net
using Lisp SBCL 1.1.17

Barton Willis <willisb@unk.edu> looked at this, too:

A workaround seems to be to declare m to be a positive integer:

~~~~
::text
(%i1) (declare(m,integer), assume(m>0))$
(%i2) integrate(exp( -x^2/2 ) / sqrt(2*%pi) * x^m, x, -1/4, 1/4)$
(%i3) %,m = 4;
(%o3) (3*sqrt(2)*sqrt(%pi)-2^(5/2)*gamma_incomplete(5/2,1/32))/(sqrt(2)*sqrt(%pi))
(%i4) float(%);
(%o4) 1.5240024317779744*10^-4
~~~~

This workaround should not be needed. Setting domain to complex makes answer to be wrong again :(

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 09:25:44
Created by **[rswarbrick](https://sourceforge.net/u/rswarbrick/)** on 2014-05-05 21:28:21
Original: https://sourceforge.net/p/maxima/bugs/2726/#a2d0

---

A simpler function that gives the wrong answer in the same way (but might be a bit easier to trace with):

~~~~
::text
(%i28) integrate (exp(-x^2) * x^m, x, -1/4, 1/4);
              m                  m + 1  1                      m + 1  1
         (- 1)  gamma_incomplete(-----, --)   gamma_incomplete(-----, --)
                                   2    16                       2    16
(%o28) - ---------------------------------- - ---------------------------
                         2                                 2
(%i29) %, m=4;
                                              5  1
(%o29)                     - gamma_incomplete(-, --)
                                              2  16
(%i30) %, numer;
(%o30)                        - 1.328966785070267
~~~~

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 09:25:48
Created by **[rswarbrick](https://sourceforge.net/u/rswarbrick/)** on 2014-05-05 21:33:18
Original: https://sourceforge.net/p/maxima/bugs/2726/#ee9b

---

After a quick glance, it looks like `integrate` is finding an antiderivative:

~~~~
::text
(%i31) integrate (exp(-x^2/2) * x^m, x);
            m + 1
            ----- - 1                          2
              2                        m + 1  x    m + 1       - m - 1
(%o31)   - 2          gamma_incomplete(-----, --) x      abs(x)
                                         2    2
~~~~

Unfortunately, that antiderivative isn't continuous at zero, which you can see with

~~~~
::text
(%i32) plot2d(%o31, [x, -1, 1]), m=4;
~~~~

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 09:25:51
Created by **[rswarbrick](https://sourceforge.net/u/rswarbrick/)** on 2014-05-05 22:33:09
Original: https://sourceforge.net/p/maxima/bugs/2726/#259a

---

A bit more information. This is in `integrate-exp-special`, after matching "Type 2". I think the problem is caused by this simplification:

~~~~
::text
(%i46) (x^2/2)^((1+m)/2);
                                        m + 1
                                  abs(x)
(%o46)                            -----------
                                     m + 1
                                     -----
                                       2
                                    2
~~~~

Obviously the simplification is correct but as `x` changes sign, we're jumping between two branches of `sqrt`. The formula for the special function integration assumes that this comes out without the absolute value signs.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 09:25:55
Created by **[rswarbrick](https://sourceforge.net/u/rswarbrick/)** on 2014-05-11 18:59:57
Original: https://sourceforge.net/p/maxima/bugs/2726/#5b79

---

That last comment was utter nonsense. Anyway, here's an even simpler example:

~~~~
::text
(%i134) integrate(x^m*%e^(-x^2), x);

Type 2: z^v*a^(b*z^r+d) : w = ((B . -1) (R . 2) (Z . $X) (D . 0) (A . $%E)
                               (V . $M) (Z . $X))
                                 m + 1   2   m + 1       - m - 1
                gamma_incomplete(-----, x ) x      abs(x)
                                   2
(%o134)       - ------------------------------------------------
                                       2
(%i135) plot2d(%, [x, -10, 10]), m=2;
plot2d: expression evaluates to non-numeric value somewhere in plotting range.
(%o135) /home/rupert/maxout.gnuplot_pipes
~~~~

I'm attaching the resulting plot, which clearly shows the discontinuity from the (x/abs(x))^(m+1) factor.

I think there's some machinery in the definite integration code to spot poles that cause this sort of thing with integrals of rational functions. Not sure whether the right thing is to come up with a more complicated formula for a continuous antiderivative, or to try to teach the definite integral code that it can't just substitute in the end points here.



Attachments:
- https://sourceforge.net/p/maxima/bugs/_discuss/thread/ae77638d/5b79/attachment/moment.png

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 09:25:58
Created by **[rswarbrick](https://sourceforge.net/u/rswarbrick/)** on 2014-05-11 19:34:15
Original: https://sourceforge.net/p/maxima/bugs/2726/#5a0e

---

This bug is fixed by 76f60fe, which I just pushed. It turns out that we've already got rather good code for spotting this sort of situation. Unfortunately, it didn't spot the possible pole of %o134 in the example above because it didn't know the sign of m.

Changing the code to be more paranoid turns out not to massively slow down integration (a nice surprise), and has the benefit of fixing this bug.
